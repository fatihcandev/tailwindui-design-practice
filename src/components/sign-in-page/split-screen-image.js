import React from 'react'
import Img from 'gatsby-image'
import { graphql, useStaticQuery } from 'gatsby'

export const PureSplitScreenImage = ({ data }) => (
  <Img fluid={data.file.childImageSharp.fluid} alt="an architecture" />
)

export const SplitScreenImage = props => {
  const data = useStaticQuery(query)

  return <PureSplitScreenImage {...props} data={data} />
}

export default SplitScreenImage

const query = graphql`
  query {
    file(relativePath: { eq: "sign-in-page-image.jpg" }) {
      childImageSharp {
        fluid(quality: 100) {
          ...GatsbyImageSharpFluid
        }
      }
    }
  }
`
