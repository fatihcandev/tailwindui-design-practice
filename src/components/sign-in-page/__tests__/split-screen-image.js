import React from 'react'
import renderer from 'react-test-renderer'
import { PureSplitScreenImage as SplitScreenImage } from '../split-screen-image'

describe('SplitScreenImage', () => {
  it('renders correctly', () => {
    const data = {
      file: {
        childImageSharp: {
          fluid: {
            base64:
              'data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAALABQDASIAAhEBAxEB/8QAGAAAAgMAAAAAAAAAAAAAAAAAAAMCBAX/xAAWAQEBAQAAAAAAAAAAAAAAAAACAQT/2gAMAwEAAhADEAAAAYPuJWcNEj//xAAaEAACAwEBAAAAAAAAAAAAAAABAgAEEQMy/9oACAEBAAEFAoi7DXba6g81GBPH/8QAGREAAgMBAAAAAAAAAAAAAAAAAAIBAxNB/9oACAEDAQE/AUr7Jkp//8QAFhEBAQEAAAAAAAAAAAAAAAAAAAER/9oACAECAQE/AbWv/8QAGhAAAwADAQAAAAAAAAAAAAAAAAERISIxYf/aAAgBAQAGPwIr4avA6ieiP//EABoQAQACAwEAAAAAAAAAAAAAAAEAERAxQbH/2gAIAQEAAT8hhwtYPFwsBt7CQaF7if/aAAwDAQACAAMAAAAQrz//xAAYEQACAwAAAAAAAAAAAAAAAAAAAREhMf/aAAgBAwEBPxBFYEh//8QAFxEBAAMAAAAAAAAAAAAAAAAAAAERQf/aAAgBAgEBPxDGF3//xAAZEAEAAwEBAAAAAAAAAAAAAAABABEhQVH/2gAIAQEAAT8QjoC+Bq4azfUqYBN1pfISqBgq9exLZuT/2Q==',
            aspectRatio: 1.7699115044247788,
            src:
              '/static/368f6b6dfb83b0b5eb3d6a7030a14213/2f1b1/sign-in-page-image.jpg',
            srcSet:
              '/static/368f6b6dfb83b0b5eb3d6a7030a14213/fd013/sign-in-page-image.jpg 200w,\n/static/368f6b6dfb83b0b5eb3d6a7030a14213/25252/sign-in-page-image.jpg 400w,\n/static/368f6b6dfb83b0b5eb3d6a7030a14213/2f1b1/sign-in-page-image.jpg 800w,\n/static/368f6b6dfb83b0b5eb3d6a7030a14213/0ff54/sign-in-page-image.jpg 1200w,\n/static/368f6b6dfb83b0b5eb3d6a7030a14213/06655/sign-in-page-image.jpg 1600w,\n/static/368f6b6dfb83b0b5eb3d6a7030a14213/097fa/sign-in-page-image.jpg 1920w',
            sizes: '(max-width: 800px) 100vw, 800px'
          }
        }
      }
    }
    const tree = renderer.create(<SplitScreenImage data={data} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
