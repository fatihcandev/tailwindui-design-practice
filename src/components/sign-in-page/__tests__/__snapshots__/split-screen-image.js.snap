// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplitScreenImage renders correctly 1`] = `
<div
  className=" gatsby-image-wrapper"
  style={
    Object {
      "maxHeight": null,
      "maxWidth": null,
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <div
    aria-hidden={true}
    style={
      Object {
        "paddingBottom": "56.5%",
        "width": "100%",
      }
    }
  />
  <img
    alt=""
    aria-hidden={true}
    src="data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAALABQDASIAAhEBAxEB/8QAGAAAAgMAAAAAAAAAAAAAAAAAAAMCBAX/xAAWAQEBAQAAAAAAAAAAAAAAAAACAQT/2gAMAwEAAhADEAAAAYPuJWcNEj//xAAaEAACAwEBAAAAAAAAAAAAAAABAgAEEQMy/9oACAEBAAEFAoi7DXba6g81GBPH/8QAGREAAgMBAAAAAAAAAAAAAAAAAAIBAxNB/9oACAEDAQE/AUr7Jkp//8QAFhEBAQEAAAAAAAAAAAAAAAAAAAER/9oACAECAQE/AbWv/8QAGhAAAwADAQAAAAAAAAAAAAAAAAERISIxYf/aAAgBAQAGPwIr4avA6ieiP//EABoQAQACAwEAAAAAAAAAAAAAAAEAERAxQbH/2gAIAQEAAT8hhwtYPFwsBt7CQaF7if/aAAwDAQACAAMAAAAQrz//xAAYEQACAwAAAAAAAAAAAAAAAAAAAREhMf/aAAgBAwEBPxBFYEh//8QAFxEBAAMAAAAAAAAAAAAAAAAAAAERQf/aAAgBAgEBPxDGF3//xAAZEAEAAwEBAAAAAAAAAAAAAAABABEhQVH/2gAIAQEAAT8QjoC+Bq4azfUqYBN1pfISqBgq9exLZuT/2Q=="
    style={
      Object {
        "height": "100%",
        "left": 0,
        "objectFit": "cover",
        "objectPosition": "center",
        "opacity": 1,
        "position": "absolute",
        "top": 0,
        "transitionDelay": "500ms",
        "width": "100%",
      }
    }
  />
  <picture>
    <source
      sizes="(max-width: 800px) 100vw, 800px"
      srcSet="/static/368f6b6dfb83b0b5eb3d6a7030a14213/fd013/sign-in-page-image.jpg 200w,
/static/368f6b6dfb83b0b5eb3d6a7030a14213/25252/sign-in-page-image.jpg 400w,
/static/368f6b6dfb83b0b5eb3d6a7030a14213/2f1b1/sign-in-page-image.jpg 800w,
/static/368f6b6dfb83b0b5eb3d6a7030a14213/0ff54/sign-in-page-image.jpg 1200w,
/static/368f6b6dfb83b0b5eb3d6a7030a14213/06655/sign-in-page-image.jpg 1600w,
/static/368f6b6dfb83b0b5eb3d6a7030a14213/097fa/sign-in-page-image.jpg 1920w"
    />
    <img
      alt="an architecture"
      loading="lazy"
      onLoad={[Function]}
      sizes="(max-width: 800px) 100vw, 800px"
      src="/static/368f6b6dfb83b0b5eb3d6a7030a14213/2f1b1/sign-in-page-image.jpg"
      srcSet="/static/368f6b6dfb83b0b5eb3d6a7030a14213/fd013/sign-in-page-image.jpg 200w,
/static/368f6b6dfb83b0b5eb3d6a7030a14213/25252/sign-in-page-image.jpg 400w,
/static/368f6b6dfb83b0b5eb3d6a7030a14213/2f1b1/sign-in-page-image.jpg 800w,
/static/368f6b6dfb83b0b5eb3d6a7030a14213/0ff54/sign-in-page-image.jpg 1200w,
/static/368f6b6dfb83b0b5eb3d6a7030a14213/06655/sign-in-page-image.jpg 1600w,
/static/368f6b6dfb83b0b5eb3d6a7030a14213/097fa/sign-in-page-image.jpg 1920w"
      style={
        Object {
          "height": "100%",
          "left": 0,
          "objectFit": "cover",
          "objectPosition": "center",
          "opacity": 0,
          "position": "absolute",
          "top": 0,
          "transition": "opacity 500ms",
          "width": "100%",
        }
      }
    />
  </picture>
  <noscript
    dangerouslySetInnerHTML={
      Object {
        "__html": "<picture><source srcset=\\"/static/368f6b6dfb83b0b5eb3d6a7030a14213/fd013/sign-in-page-image.jpg 200w,
/static/368f6b6dfb83b0b5eb3d6a7030a14213/25252/sign-in-page-image.jpg 400w,
/static/368f6b6dfb83b0b5eb3d6a7030a14213/2f1b1/sign-in-page-image.jpg 800w,
/static/368f6b6dfb83b0b5eb3d6a7030a14213/0ff54/sign-in-page-image.jpg 1200w,
/static/368f6b6dfb83b0b5eb3d6a7030a14213/06655/sign-in-page-image.jpg 1600w,
/static/368f6b6dfb83b0b5eb3d6a7030a14213/097fa/sign-in-page-image.jpg 1920w\\" sizes=\\"(max-width: 800px) 100vw, 800px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 800px) 100vw, 800px\\" srcset=\\"/static/368f6b6dfb83b0b5eb3d6a7030a14213/fd013/sign-in-page-image.jpg 200w,
/static/368f6b6dfb83b0b5eb3d6a7030a14213/25252/sign-in-page-image.jpg 400w,
/static/368f6b6dfb83b0b5eb3d6a7030a14213/2f1b1/sign-in-page-image.jpg 800w,
/static/368f6b6dfb83b0b5eb3d6a7030a14213/0ff54/sign-in-page-image.jpg 1200w,
/static/368f6b6dfb83b0b5eb3d6a7030a14213/06655/sign-in-page-image.jpg 1600w,
/static/368f6b6dfb83b0b5eb3d6a7030a14213/097fa/sign-in-page-image.jpg 1920w\\" src=\\"/static/368f6b6dfb83b0b5eb3d6a7030a14213/2f1b1/sign-in-page-image.jpg\\" alt=\\"an architecture\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
      }
    }
  />
</div>
`;
